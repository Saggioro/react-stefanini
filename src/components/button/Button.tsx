import React, { FunctionComponent } from 'react';

import { Container } from './Button.styled';

interface IButton extends React.HTMLAttributes<HTMLButtonElement>{
    color?: string;
    type?: 'submit' | 'button';
}

const Button:FunctionComponent<IButton> = ({color, type = 'submit', children, ...props}:IButton) => {
    return (
        <Container color={color} type={type}>
            {children}
        </Container>
    );
};

export default Button;