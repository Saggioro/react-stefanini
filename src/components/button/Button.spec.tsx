/* eslint-disable testing-library/no-debugging-utils */
import renderer from 'react-test-renderer';
import Button from "./Button";

test("Button should render correctly", () => {
  const component = renderer.create(<Button>Save</Button>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
