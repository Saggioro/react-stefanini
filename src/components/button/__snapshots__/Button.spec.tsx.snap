// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="sc-bczRLJ eVNmrq"
  type="submit"
>
  Save
</button>
`;
